### PR TITLE
IO::Buffer improvements and documentation.

### PR DESCRIPTION
**Pull request by ioquatix**
Tidy up the documentation of the C implementation.

### CODE REVIEW

**Comment by nobu** [comment]
Doesn't look documentation only?

**Comment by ioquatix** [comment]
You are correct, I reorganised and made some minor improvements of the code too.

File: io_buffer.c [hunk 1]
```diff
@@ -287,7 +294,23 @@ static const rb_data_type_t rb_io_buffer_type = {
     .flags = RUBY_TYPED_FREE_IMMEDIATELY | RUBY_TYPED_WB_PROTECTED | RUBY_TYPED_EMBEDDABLE,
 };
 
-// Extract an offset argument, which must be a positive integer.
+static inline enum rb_io_buffer_flags
+io_buffer_extract_flags(VALUE argument)
+{
+    if (rb_int_negative_p(argument)) {
+        rb_raise(rb_eArgError, "Flags can't be negative!");
```
Comment (nobu) [inline io_buffer.c:301]: We usually don’t end error messages with exclamation (or period).

Comment (ioquatix) [reply io_buffer.c:303]: Have we documented this anywhere?

Comment (ioquatix) [reply io_buffer.c:303]: Maybe we can reference something like this: https://learn.microsoft.com/en-us/dotnet/standard/exceptions/best-practices-for-exceptions

> Use grammatically correct error messages
> Write clear sentences and include ending punctuation. Each sentence in the string assigned to the [Exception.Message](https://learn.microsoft.com/en-us/dotnet/api/system.exception.message#system-exception-message) property should end in a period. For example, "The log table has overflowed." would be an appropriate message string.

**Pull request merged**